text/0000-global-asm.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: global_asm
+- Start Date: 2016-03-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC exposes LLVM's support for [module-level inline assembly](http://llvm.org/docs/LangRef.html#module-level-inline-assembly) by adding a `global_asm!` macro. The syntax is very simple: it just takes a string literal containing the assembly code.
+
+Example:
+```rust
+global_asm!(r#"
+.globl my_asm_func
+my_asm_func:
+    ret
+"#);
+
+extern {
+    fn my_asm_func();
+}
+```
+
+# Motivation
+[motivation]: #motivation
+
+There are two main use cases for this feature. The first is that it allows functions to be written completely in assembly, which mostly eliminates the need for a `naked` attribute. This is mainly useful for function that use a custom calling convention, such as interrupt handlers.
+
+Another important use case is that it allows external assembly files to be used in a Rust module without needing hacks in the build system:
```

Comment:
You need an external assembler. Where is it? Is it the right version? Does it support the target platform(s)? Does it output in the proper location? etc etc. All this makes for ugly `build.rs` boilerplate, and worse incremental rebuilds. 

This aspect of the motivation is exactly the same as that for naked functions.